File: app/src/main/java/com/marceljurtz/lifecounter/views/Settings/ISettingsView.java
package com.marceljurtz.lifecounter.views.Settings;

import com.marceljurtz.lifecounter.contracts.base.IView;
import com.marceljurtz.lifecounter.models.Color;
import com.marceljurtz.lifecounter.enums.MagicColorEnum;

public interface ISettingsView extends IView {
    // Lifepoints and Longclickpoints Getter & Setter
    void setLifepoints(String lifepointText);
    String getLifepoints();
    void setLongClickPoints(String longClickPointText);
    String getLongClickPoints();

    void returnToPrevActivity();
    void loadActivity(Class c);

    void loadResetConfirmationDialog();
    void loadColorPickerDialog(MagicColorEnum color, int r, int g, int b);
    void updateColorButtonValue(Color color);

    void setKeepScreenOnCheckbox(boolean checked);
    void setConfirmGameResetCheckbox(boolean checked);
}

File: app/src/main/java/com/marceljurtz/lifecounter/views/Settings/ISettingsView.kt
package com.marceljurtz.lifecounter.views.Settings

import com.marceljurtz.lifecounter.contracts.base.IView
import com.marceljurtz.lifecounter.models.Color
import com.marceljurtz.lifecounter.enums.MagicColorEnum

interface ISettingsView : IView {
    // Lifepoints and Longclickpoints Getter & Setter
    var lifepoints: String
    var longClickPoints: String

    fun returnToPrevActivity()
    override fun loadActivity(c: Class<*>)

    fun loadResetConfirmationDialog()
    fun loadColorPickerDialog(color: MagicColorEnum, r: Int, g: Int, b: Int)
    fun updateColorButtonValue(color: Color)

    fun setKeepScreenOnCheckbox(checked: Boolean)
    fun setConfirmGameResetCheckbox(checked: Boolean)
}
